frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx

```diff
@@ -0,0 +1,287 @@
+        const options = [
+            ['event', 'The entire event payload as a JSON object'],
+            ['event.name', 'The name of the event e.g. $pageview'],
+            ['event.distinct_id', 'The distinct_id of the event'],
+            ['event.timestamp', 'The timestamp of the event'],
+            ['event.url', 'URL to the event in PostHog'],
+            ['event.properties', 'Properties of the event'],
+            ['event.properties.<key>', 'The individual property of the event'],
+            ['person', 'The entire person payload as a JSON object'],
+            ['project.uuid', 'The UUID of the Person in PostHog'],
+            ['person.url', 'URL to the person in PostHog'],
+            ['person.properties', 'Properties of the person'],
+            ['person.properties.<key>', 'The individual property of the person'],
+            ['project.id', 'ID of the project in PostHog'],
+            ['project.name', 'Name of the project'],
+            ['project.url', 'URL to the project in PostHog'],
+            ['source.name', 'Name of the source of this message'],
+            ['source.url', 'URL to the source of this message in PostHog'],
+        ]
+
+        groupTypes.forEach((groupType) => {
+            options.push([`groups.${groupType.group_type}`, `The entire group payload as a JSON object`])
+            options.push([`groups.${groupType.group_type}.name`, `Display name of the group`])
+            options.push([`groups.${groupType.group_type}.url`, `URL to the group in PostHog`])
+            options.push([`groups.${groupType.group_type}.properties`, `Properties of the group`])
+            options.push([`groups.${groupType.group_type}.properties.<key>`, `The individual property of the group`])
+            options.push([`groups.${groupType.group_type}.index`, `Index of the group`])
+        })
+
+        const items: languages.CompletionItem[] = options.map(([key, value]) => {
+            return {
+                label: key,
+                kind: languages.CompletionItemKind.Variable,
+                detail: value,
+                insertText: key,
+                range: {
+                    startLineNumber: 1,
+                    endLineNumber: 1,
+                    startColumn: 0,
+                    endColumn: 0,
+                },
+            }
+        })
+
+        return items
+    }, [groupTypes])
+}
+
+function JsonConfigField(props: {
+    onChange?: (value: string) => void
+    className: string
+    autoFocus: boolean
+    value?: string | object
+}): JSX.Element {
+    const suggestions = useAutocompleteOptions()
+    const [monaco, setMonaco] = useState<Monaco>()
+
+    useEffect(() => {
+        if (!monaco) {
+            return
+        }
+        monaco.languages.setLanguageConfiguration('json', {
+            wordPattern: /[a-zA-Z0-9_\-.]+/,
+        })
+
+        const provider = monaco.languages.registerCompletionItemProvider('json', {
+            triggerCharacters: ['{', '{{'],
```

Comment:
Is `{{` still used for something other than escaping a `{`?

Reply:
I actually still need to implement this so that double brackets do the whole object, and single ones only a template string 🤔

Reply:
Yup... though for me this can just be in a followup. There's the "source code" escape hatch anyway.

Reply:
Added to follow up